Sync/ContactDataExchange.php

```diff
@@ -179,6 +179,11 @@ public function getObjectSyncReport(
             $objectDAO = new ObjectDAO(self::OBJECT_NAME, $object->getId(), new \DateTimeImmutable($object->getModifiedTime()->format('r')));
 
             foreach ($object->dehydrate($mappedFields) as $field => $value) {
+                // The 'reference' key doesn't seem to exist every time. Based on readme it's used only for owners. Skip it.
```

Comment:
If we're looking only for the `reference` field to skip, why are we skipping all not-set object fields?

Other than that, it's perfectly fine from a code-review perspective.

Reply:
The reference field is not set. If any of the keys is not set it will throw the notice and right after that it will throw an error that null is provided instead of the requested object and the sync will stop. Any key should be skipped if it does not exist.